lib/android-helpers.js

```diff
@@ -372,32 +373,57 @@ helpers.pushStrings = async function (language, adb, opts) {
   return {};
 };
 
-helpers.unlock = async function (adb) {
+helpers.unlock2 = async function (driver, adb, capabilities) {
+  let unlockType = capabilities.unlockType;
+  if (!unlocker.isValidUnlockType(unlockType)) {
+    throw new Error(`Invalid unlock type ${unlockType}`);
+  }
+  let unlockKey = capabilities.unlockKey;
+  if (!unlocker.isValidKey(unlockType, unlockKey)) {
+    throw new Error(`Missing unlockKey ${unlockKey} capability for unlockType ${unlockType}`);
+  }
+  let unlockMethod;
+  if (unlockType === unlocker.PIN_UNLOCK) {
```

Comment:
UPD: It seems like JS can also handle string constants as _switch_ arguments: http://stackoverflow.com/questions/2896626/switch-statement-for-string-matching-in-javascript